packages/core-utils/src/itinerary.js

```diff
@@ -521,6 +517,35 @@ export function calculateFares(itinerary) {
     }
   });
 
+  if (multiple) {
+    // Return object of fares
+    const transitFares = {};
+    if (itinerary && itinerary.fare && itinerary.fare.fare) {
+      Object.keys(itinerary.fare.fare).forEach(fareKey => {
+        const fareComponent = itinerary.fare.fare[fareKey];
+        transitFares[fareKey] = getTransitFare(fareComponent);
+      });
+    }
+
+    return {
+      maxTNCFare,
+      minTNCFare,
+      tncCurrencyCode,
+      transitFares
+    };
+  }
+
+  // Extract fare total from itinerary fares.
+  const fareComponent =
+    itinerary.fare && itinerary.fare.fare && itinerary.fare.fare.regular;
```

Comment:
Add FIXME for specifying what the "default" fare component in cases it is not `regular`.

Reply:
Large comment added in 7f5f94945d6ceae3eaf1feeaf966f92b92dba660